rust/datafusion/tests/sql.rs

```diff
@@ -1941,6 +1941,59 @@ async fn boolean_expressions() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn interval_expressions() -> Result<()> {
+    let mut ctx = ExecutionContext::new();
+    let sql = "SELECT
+        (interval '1') as interval_1,
+        (interval '1 second') as interval_2,
+        (interval '500 milliseconds') as interval_3,
+        (interval '5 second') as interval_4,
+        (interval '1 minute') as interval_5,
+        (interval '0.5 minute') as interval_6,
+        (interval '.5 minute') as interval_7,
+        (interval '5 minute') as interval_8,
+        (interval '5 minute 1 second') as interval_9,
+        (interval '1 hour') as interval_10,
+        (interval '5 hour') as interval_11,
+        (interval '1 day') as interval_12,
+        (interval '5 day') as interval_13,
+        (interval '5 day 4 hours 3 minutes 2 seconds 100 milliseconds') as interval_14,
+        (interval '1 month') as interval_15,
+        (interval '5 month') as interval_16,
+        (interval '13 month') as interval_17,
+        (interval '1 year') as interval_18,
+        (interval '2 year') as interval_19,
+        (interval '1 day 1') as interval_20
```

Comment:
might be worth testing with a fractional interval as well `interval '2.3' years` as I think your implementation parses the numbers as a `f32`

Reply:
Thanks, added.